### PR TITLE
Use <For /> in Heading documentation

### LINKED ISSUE #3705: Use `<For />` in `Heading` documentation
### Provide the exact quote of the error or issue
We should use `<For />` component when repeating a component
e.g.
```tsx
<For each={items}>{(item, index) => <Text key={index} fontWeight="bold">{item}</Text>}</For>
```

https://yamada-ui.com/components/typography/heading

### Does the docs page already exist? Please link to it.

_No response_

### Are you willing to participate in fixing this issue and create a pull request with the fix?

None

### PR DESCRIPTION
**Pull request by illionillion**
Closes #3705  

## Description

Fixed the repetition part in the `Heading` documentation to use the `For` component.

### CODE REVIEW

File: docs/contents/components/typography/heading/index.ja.mdx [hunk 1]
```diff
@@ -22,40 +22,28 @@ import { Heading } from "@yamada-ui/react"
...
-  <Heading as="h6" size="xs" isTruncated>
-    ギャルのパンティーおくれーーーっ！！！！！
-  </Heading>
-</VStack>
+```tsx functional=true
+const items = [
+  { as: "h1", size: "4xl" },
+  { as: "h2", size: "3xl" },
+  { as: "h2", size: "2xl" },
+  { as: "h3", size: "xl" },
```
Comment (chida09) [inline docs/contents/components/typography/heading/index.ja.mdx:30]: @illionillion
The `as="h3" size="lg"` specification has been removed - is this intentional?

Comment (illionillion) [reply docs/contents/components/typography/heading/index.ja.mdx:30]: @chida09 
That was my mistake.
Thanks for pointing that out.

**Comment by illionillion** [comment]
@chida09 
Please review again.

**Review by chida09 — DISMISSED** [review]

File: docs/contents/components/typography/heading/index.mdx [hunk 2]
```diff
@@ -24,40 +24,29 @@ import { Heading } from "@yamada-ui/react"
...
+  { as: "h6", size: "xs" },
+]
 
-  <Heading as="h6" size="xs" isTruncated>
-    ギャルのパンティーおくれーーーっ！！！！！
-  </Heading>
-</VStack>
+return (
+  <VStack>
+    <For each={items}>
```
Comment (hirotomoyamada) [inline docs/contents/components/typography/heading/index.mdx:41]: Change

```
<For each={items}>
```

to

```
<For
  each={[
    { as: "h1", size: "4xl" },
    { as: "h2", size: "3xl" },
    { as: "h2", size: "2xl" },
    { as: "h3", size: "xl" },
    { as: "h3", size: "lg" },
    { as: "h4", size: "md" },
    { as: "h5", size: "sm" },
    { as: "h6", size: "xs" },
  ]}
>
```

File: docs/contents/components/typography/heading/index.ja.mdx [hunk 3]
```diff
@@ -22,40 +22,29 @@ import { Heading } from "@yamada-ui/react"
...
+  { as: "h3", size: "lg" },
+  { as: "h4", size: "md" },
+  { as: "h5", size: "sm" },
+  { as: "h6", size: "xs" },
+]
+
+return (
+  <VStack>
+    <For each={items}>
```
Comment (hirotomoyamada) [inline docs/contents/components/typography/heading/index.ja.mdx:39]: Change

```
<For each={items}>
```

to

```
<For
  each={[
    { as: "h1", size: "4xl" },
    { as: "h2", size: "3xl" },
    { as: "h2", size: "2xl" },
    { as: "h3", size: "xl" },
    { as: "h3", size: "lg" },
    { as: "h4", size: "md" },
    { as: "h5", size: "sm" },
    { as: "h6", size: "xs" },
  ]}
>
```

**Comment by hirotomoyamada** [comment]
@illionillion @chida09 

Documentation would like to have consistent code descriptions. I would like it to be as per the comments in the review.

File: docs/contents/components/typography/heading/index.ja.mdx [hunk 4]
```diff
@@ -22,40 +22,29 @@ import { Heading } from "@yamada-ui/react"
...
-
-  <Heading as="h5" size="sm" isTruncated>
-    ギャルのパンティーおくれーーーっ！！！！！
-  </Heading>
-
-  <Heading as="h6" size="xs" isTruncated>
-    ギャルのパンティーおくれーーーっ！！！！！
-  </Heading>
-</VStack>
+```tsx functional=true
```
Comment (hirotomoyamada) [inline docs/contents/components/typography/heading/index.ja.mdx:25]: Remove `functional=true`.

File: docs/contents/components/typography/heading/index.mdx [hunk 5]
```diff
@@ -24,40 +24,29 @@ import { Heading } from "@yamada-ui/react"
...
-  </Heading>
-
-  <Heading as="h4" size="md" isTruncated>
-    ギャルのパンティーおくれーーーっ！！！！！
-  </Heading>
-
-  <Heading as="h5" size="sm" isTruncated>
-    ギャルのパンティーおくれーーーっ！！！！！
-  </Heading>
+```tsx functional=true
```
Comment (hirotomoyamada) [inline docs/contents/components/typography/heading/index.mdx:27]: Remove `functional=true`.

**Comment by illionillion** [comment]
@hirotomoyamada @chida09 I fixed those. Plase review again!

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**